lib/thrift/servers/binary_framed/server.ex

```diff
@@ -0,0 +1,31 @@
+defmodule Thrift.Servers.BinaryFramed do
+  @moduledoc false
+
+  @type server_opts :: [
+    worker_count: pos_integer
+  ]
+
+  alias Thrift.Servers.BinaryFramed.AcceptorSupervisor
+
+  defmodule State do
+    @moduledoc false
+    defstruct port: nil, socket: nil, supervisor: nil
+  end
+
+  @type opts :: [
+    port: (1..65535),
+  ]
+
+  @spec start_link(module, (1..65535), module, server_opts, GenServer.option) :: GenServer.on_start
+  def start_link(server_module, port, handler_module, opts, gen_server_opts) do
+    GenServer.start_link(__MODULE__, [server_module, port, handler_module, opts], gen_server_opts)
+  end
+
+  def init([server_module, port, handler_module, opts]) do
+    worker_count = Keyword.get(opts, :worker_count, 1)
+
+    {:ok, socket} = :gen_tcp.listen(port, [:binary, packet: 4, active: false])
+    {:ok, supervisor} = AcceptorSupervisor.start_link(socket, server_module, handler_module, worker_count)
```

Comment:
With the current design I think this could be merged with `AcceptorSupervisor` as a single process/supervisor. This would ensure that lower down in the supervision tree (`AcceptorSupervisor` and its descendants) are terminated before the process in this position in the tree exits.

@pguillory here is an example where they are split: https://github.com/fishcakez/acceptor_pool/tree/master/examples/telnet_echo (requires OTP 19).